projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/components/seo-assistant/index.tsx

```diff
@@ -1,21 +1,47 @@
+import { useModuleStatus } from '@automattic/jetpack-shared-extension-utils';
 import { Button } from '@wordpress/components';
+import { useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import debugFactory from 'debug';
+import { SeoPlaceholder } from '../../../../plugins/seo/components/placeholder';
+import usePostContent from '../../hooks/use-post-content';
+import './style.scss';
+import bigSkyIcon from './big-sky-icon.svg';
+import SeoAssistantWizard from './seo-assistant-wizard';
+import type { SeoAssistantProps } from './types';
 
 const debug = debugFactory( 'jetpack-ai:seo-assistant' );
 
-export default function SeoAssistant( { busy, disabled } ) {
+export default function SeoAssistant( { disabled, onStep }: SeoAssistantProps ) {
+	const [ isOpen, setIsOpen ] = useState( false );
+	const postContent = usePostContent();
+	const { isLoadingModules, isChangingStatus, isModuleActive, changeStatus } =
+		useModuleStatus( 'seo-tools' );
+
+	debug( 'rendering seo-assistant entry point' );
 	return (
 		<div>
 			<p>{ __( 'Improve post engagement.', 'jetpack' ) }</p>
-			<Button
-				onClick={ () => debug( 'click' ) }
-				variant="secondary"
-				disabled={ disabled }
-				isBusy={ busy }
-			>
-				{ __( 'SEO Assistant', 'jetpack' ) }
-			</Button>
+			{ ( isModuleActive || isLoadingModules ) && (
+				<Button
+					onClick={ () => setIsOpen( true ) }
+					variant="secondary"
+					disabled={ isLoadingModules || isOpen || ! postContent.trim?.() || disabled }
```

Comment:
This makes `usePostContent` be called repeatedly. We can remove this from the `disabled` condition and check only on `onClick`.